okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -282,6 +312,12 @@ void streamReadyToStart(OkHttpClientStream clientStream) {
   private void startStream(OkHttpClientStream stream) {
     Preconditions.checkState(stream.id() == null, "StreamId already assigned");
     streams.put(nextStreamId, stream);
+    if (keepAliveManager != null) {
+      // We have a new stream. We might need to do keepalives now.
+      // Note that we have to do this inside the lock to avoid calling
+      // KeepAliveManager.onTransportActive and KeepAliveManager.onTransportIdle in the wrong order.
+      keepAliveManager.onTransportActive();
```

Comment:
This can be done in `setInUse()` instead, before calling `listener.transportInUse(true)`.